tracing-core/src/span.rs

```diff
@@ -195,6 +195,12 @@ impl<'a> Attributes<'a> {
     pub fn is_empty(&self) -> bool {
         self.values.is_empty()
     }
+
+    /// Returns the number of fields in this `Attributes`.
+    #[inline]
+    pub fn len(&self) -> usize {
+        self.values.field_set().len()
+    }
```

Comment:
I'm open to adding this in `tracing-core`, but note that it means that `tracing-opentelemetry` releases that include this change on this will need to increase their minimum `tracing-core` version to include this new API.

As an alternative, can we just use `attributes.values().field_set().len()` in `tracing-opentelemetry`, so that it doesn't depend on the `tracing-core` change? We can add a `len` method to `Attributes` as well in a separate branch.

This isn't a huge deal, but it might be better.

Reply:
> As an alternative, can we just use `attributes.values().field_set().len()` in `tracing-opentelemetry`, so that it doesn't depend on the `tracing-core` change?

Unfortunately, we can't do this right now since the `field_set()` is `pub(crate)`.
```rust
    pub(crate) fn field_set(&self) -> &FieldSet {
        self.fields
    }
```

I think we should submit another PR adding `len()` method to `tracing-core` and bump the minor version. Then this PR should depend on the new minor version. What do you think?

Reply:
> > 
> 
> Unfortunately, we can't do this right now since the `field_set()` is `pub(crate)`.
> 
> ```rust
> 
> ```

We can also [get the `FieldSet` from the span's metadata](https://docs.rs/tracing-core/0.1.17/tracing_core/metadata/struct.Metadata.html#method.fields); this is part of the public API. So, if we change this PR to use 
```rust
attributes.metadata().fields().len()
```
we should be able to merge a separately from the `tracing-core` change.

What do you think?

Reply:
Good idea. 👍